graphql/executor.go

```diff
@@ -158,7 +170,17 @@ func (e *Executor) resolveAndExecute(ctx context.Context, field *Field, source i
 			}
 
 			// TODO: Consider cacheing resolve and execute independently
+			fromCache := true
+			defer func() {
+				if span != nil {
+					span.LogFields(
+						log.Bool("cached", fromCache),
+					)
+					span.Finish()
```

Comment:
It would be good to comment why you call span.Finish here and not from a defer right after you add the span to the context.